mindmeld/query_cache.py

```diff
@@ -43,6 +44,10 @@ def copy_db(source, dest):
 
     def __init__(self, app_path):
         # make generated directory if necessary
+        text_preparation_pipeline = TextPreparationPipelineFactory.create_from_app_path(
```

Comment:
Are there any side effects to instantiating a TextPreparationPipelineFactory that we should be aware of here?

Reply:
Only one edge case related to instantiating another TextPreparationPipeline is if a developer creates a custom Application object by passing in a TextPreparationPipeline object directly, this would propagate all the way down to the QueryFactory. By instantiating a separate TextPreparationPipeline here we would not be taking into account a custom object passed in from the top-level.

However, this is a very specific edge case as it requires to create a custom Application using a pre-instantiated TextPreparationPipeline and add new data.

Reply:
So is it relatively easy to pass in the TextPreparationPipeline object instead of instantiating it here?  That could also make it easier to test this class.

Reply:
If there are class methods to grab the existing object of the Application, Resource Loader or Query Factory then you can get the object through that (assuming that they've been implemented as Singleton classes).

Reply:
Refactored code to pass TextPreparationPipeline into QueryCache via QueryFactory. Nice catch @davidmfrey @KKsharma99 !